src/graph/graph.c

```diff
@@ -503,6 +514,21 @@ void Graph_GetEdgesConnectingNodes(const Graph *g, NodeID srcID, NodeID destID,
 	}
 }
 
+// Static function to add node to label matrix - used in Graph_CreateNode and Graph_SetNode.
+static void _Graph_AddNodeToLabelMatrix(Graph *g, NodeID id, int label) {
```

Comment:
I suggest inverting this into a guard clause:
`if(label == GRAPH_NO_LABEL) return;`